api/build.gradle

```diff
@@ -0,0 +1,21 @@
+plugins {
+    id 'java-library'
+    id 'kotlin'
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_7
+    targetCompatibility = JavaVersion.VERSION_1_7
+}
+
+dependencies{
+    implementation "androidx.core:core-ktx:$core_ktx_version"
+    implementation "com.squareup.retrofit2:retrofit:$retrofit_version"
+    implementation "com.squareup.okhttp3:logging-interceptor:$okhttp_version"
+    implementation "com.squareup.okhttp3:okhttp:$okhttp_version"
+    implementation "com.squareup.retrofit2:converter-gson:$gson_converter_version"
+    implementation "com.google.code.gson:gson:$gson_version"
+
+    implementation project(':core')
+    implementation project(':dataSource')
```

Comment:
It's totally wrong to provide datasource module to API module. These modules should 'know' nothing about each other. remove  `implementation project(':dataSource')`

Reply:
So how the API will use the model which we download from the server ?